File: src/main/java/com/zndevs/toolset/tools/URLDownloader.java
package com.zndevs.toolset.tools;

import java.io.*;
import java.net.URL;
import java.net.URLConnection;

/**
 * URL Downloader
 */

public class URLDownloader {

    private String data = "";

    public URLDownloader(String urlString) throws IOException {
        data = downloadUrl(urlString);
    }

    public URLDownloader() {
    }

    public String downloadUrl(String urlString) throws IOException {
        URL url = new URL(urlString);
        URLConnection connection = url.openConnection();

        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));

        String inputLine;
        StringBuilder totalContents = new StringBuilder();
        while ((inputLine = reader.readLine()) != null) {
            totalContents.append(inputLine).append("\n");
        }
        reader.close();

        return data = totalContents.toString();
    }

    public String getResponse() {
        return this.data;
    }

    public byte[] downloadUrlBytes(String urlString) throws IOException {
        InputStream is = null;
        URL url = new URL(urlString);
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        try {
            is = url.openStream();
            byte[] chunk = new byte[4096];
            int n;

            while ((n = is.read(chunk)) > 0) {
                baos.write(chunk, 0, n);
            }
        } finally {
            if (is != null) is.close();
        }

        return baos.toByteArray();
    }

}

File: src/main/kotlin/com/zndevs/toolset/tools/URLDownloader.kt
package com.zndevs.toolset.tools

import com.sun.deploy.net.HttpDownload
import sun.net.www.http.HttpClient
import java.io.*
import java.net.URL

/**
 * URL Downloader
 */

class URLDownloader {

    var response = ""
        private set

    constructor(urlString: String) {
        response = downloadUrl(urlString)
    }

    constructor()

    fun downloadUrl(urlString: String): String {
        val url = URL(urlString)
        val connection = url.openConnection()

        val inputStream = connection.getInputStream()

        val response = inputStream.bufferedReader().use { it.readText() }
        return response
    }

    fun downloadUrlBytes(urlString: String): ByteArray {
        val url = URL(urlString)

        url.openStream().use {
            return it.readBytes()
        }
    }

}
